# Profitability of a single PV system; Decline of revenues of a distribution system grid due to increased PV self-generation

In [1]:
#given data
#PV-system
C_installed = 3 #kWp
P_invest_spec = 800 #€/kWp
lifetime = 25 #years
P_invest = P_invest_spec*C_installed

#electricity purchase and sale
P_supply = 0.2 # €/kWh
P_grid_var = 0.05 # €/kWh
P_grid_fixed = 100 # €/a
R_sales = 0.1 #€/kWh

#profiles
load_d = 5*0.5+1+1.5*3+2*2+3*2+2.5*2+2+1.5*2+2*2+1.5+0.5*3 #kWh
load_a = load_d*365
generation_d = 0.5*2+2+2.5*2+3*2+2.5*2+2+1.5+0.5 #kWh
generation_a = generation_d*365
#grid_d = 0.5*5+1+1.5+2*1+1.5+2+1.5+3*0.5 #kWh
grid_d = 0.5*5+1+1.5+2*1+0.5+1.5+2+1.5+3*0.5 #kWh !!!Ausgebesserter Fehler!!!
grid_a = grid_d*365
overproduction_d = 0.5+0.5+0.5+0.5 #kWh
overproduction_a = overproduction_d*365

## a) Net Present Value

In [2]:
AF = 0.070 #see table: 25 year, 5% interest rate)
interest_rate = 0.05
P_1 = (grid_a*P_supply)+(grid_a*P_grid_var)+P_grid_fixed
R_1 = overproduction_a*R_sales

# NPV for the PV-system
npv_1 = -P_invest  

for t in range (1, lifetime + 1):
    npv_1 -= (P_1)/(1+interest_rate)**t
    npv_1 += R_1/(1+interest_rate)**t


print("Net Present Value over the entire technical lifetime of the PV system is",round(npv_1,2))

Net Present Value over the entire technical lifetime of the PV system is -20785.55


In [3]:
# NPV for purchase only
P_2 = (load_a*P_supply)+(load_a*P_grid_var)+P_grid_fixed
print(P_2)
R_2 = 0

npv_2 = -0  # no PV installed

for t in range (1, lifetime + 1):
    npv_2 -= P_2/(1+interest_rate)**t
    npv_2 += R_2/(1+interest_rate)**t
print("Net Present Value from pure electricity purchase with no pv installed is",round(npv_2,2))

P_2a = (load_a*P_supply)+(load_a*P_grid_var)+P_grid_fixed
R_2a = generation_a*R_sales

npv_2a = -P_invest  # no PV installed

for t in range (1, lifetime + 1):
    npv_2a -= P_2a/(1+interest_rate)**t
    npv_2a += R_2a/(1+interest_rate)**t
print("Net Present Value from pure electricity purchase with pv installed is",round(npv_2a,2))

3293.75
Net Present Value from pure electricity purchase with no pv installed is -46421.93
Net Present Value from pure electricity purchase with pv installed is -36990.06


In [4]:
# NPV with no remuneration for surplus feedin
P_3 = (grid_a*P_supply)+(grid_a*P_grid_var)+P_grid_fixed
print(P_3)
R_3 = 0

npv_3 = -P_invest

for t in range (1, lifetime + 1):
    npv_3 -= P_3/(1+interest_rate)**t
    npv_3 += R_3/(1+interest_rate)**t
print("Net Present Value if there is no remuneration for surplus feed-in is",round(npv_3,2))

1377.5
Net Present Value if there is no remuneration for surplus feed-in is -21814.41


#### Interpretation: With a PV-system the costs for electricity shrink to less than half of the original value. In this chase not getting any remuneration is barely affecting the calculation, meaning that most of the benefit is achivied by the self-consumption of generated energy.

## b) Partial shading and financial compensation

In [5]:
#profiles:
load_d_new = 5*0.5+1+1.5*3+2*2+3*2+2.5*2+2+1.5*2+2*2+1.5+0.5*3 #kWh
load_a_new = load_d_new*365
generation_d_new = 0.5+2+2.5*2+3*2+2.5*2+1+2*0.5 #kWh !!! Fehler ausgebessert!!!
generation_a_new = generation_d_new*365
grid_d_new = 0.5*5+1+1.5*2+1+0.5+0.5+1+1.5+2+1.5+0.5*3 #kWh !!! Fehler ausgebessert!!!
grid_a_new = grid_d_new*365
overproduction_d_new = 0.5+0.5+0.5+0.5 #kWh !!! Fehler ausgebessert!!!
overproduction_a_new = overproduction_d_new*365

In [6]:
P_new = (grid_a_new*P_supply)+(grid_a_new*P_grid_var)+P_grid_fixed
R_new = overproduction_a_new*R_sales

# NPV for the PV-system
npv_new = -P_invest  # Initial investment cost

for t in range (1, lifetime + 1):
    npv_new -= P_new/(1+interest_rate)**t
    npv_new += R_new/(1+interest_rate)**t

print("Net Present Value over the entire technical lifetime of the PV system is",round(npv_new,2))
print("Difference between a) and b) is",round((npv_1-npv_new),2))

Net Present Value over the entire technical lifetime of the PV system is -23357.7
Difference between a) and b) is 2572.14


#### Interpretation: Comparing this to the situation in a) the system operator looses 2572.14 € in profit - which the property developer has to compensate.

## c) Decline in reveneues for the grid operator for 1000 households

In [7]:
load_a_1000ho = load_a*1000
grid_a_1000ho = grid_a*1000

revenue_nopv = load_a_1000ho*P_grid_var
revenue_pv = grid_a_1000ho*P_grid_var

revenue_dec = revenue_nopv-revenue_pv

increase_fixed_grid_t = revenue_dec/10000

print('The decline of the distribution grid operator revenues is:', round((revenue_dec),2), '€, \nin order to compensate the grid operator, the fixed grid tariff for 10000 households must be increased by', round(increase_fixed_grid_t,2), '€ ,  \nwich leads to a total of the fixed grid tariff of', round(P_grid_fixed + increase_fixed_grid_t,2),'€/a')



The decline of the distribution grid operator revenues is: 383250.0 €, 
in order to compensate the grid operator, the fixed grid tariff for 10000 households must be increased by 38.33 € ,  
wich leads to a total of the fixed grid tariff of 138.32 €/a


## d) Penalties for providing to much electricity in the grid

In [8]:
load_d_pd = 0.5*5+1+1.5*3+2*2+2.5*3+2*2+1.5*2+ 2.5+3+2.5+1.5+2*0.5 #kwh
generation_d_pd = 2*0.5+2+2*2.5+3+3.5+4+3+2.5+1.5+0.5 #kwh
grid_d_pd = 0.5*5+1+1.5+2*1+2+3+2.5+1.5+2*0.5 #kwh
overproduction_d_pd= 0.5+0.5+1+2+1+1 #kwh
overproduction_d_pd_pen = 2 #kwh

feedin_pen = 5 #€/kw

load_a_pd = load_d * 355 + load_d_pd*10
generation_a_pd = generation_d *355+ generation_d_pd *10
grid_a_pd = grid_d * 355+ grid_d_pd * 10
overproduction_a_pd = overproduction_d*355+overproduction_d_pd*10
overproduction_a_pd_pen = overproduction_d_pd_pen *10

P_pd = (grid_a*P_supply)+(grid_a*P_grid_var)+P_grid_fixed
P_pen_pd = overproduction_a_pd_pen*feedin_pen
R_pd = overproduction_a*R_sales

# NPV for the PV-system
npv_pd = -P_invest  

for t in range (1, lifetime + 1):
    npv_pd -= P_pd/(1+interest_rate)**t
    npv_pd -= P_pen_pd/(1+interest_rate)**t
    npv_pd += R_pd/(1+interest_rate)**t

print("Net Present Value over the entire technical lifetime of the PV system with the penalty is",round(npv_pd,2))

Net Present Value over the entire technical lifetime of the PV system with the penalty is -22194.95


As you can see the NPV is more negative in point d than in a! It would be recommended to increase the load in the the times where to much power is fed in the grid -> Higher self consumption.
